upstairs/src/lib.rs

```diff
@@ -941,22 +942,40 @@ where
                             up.encrypted(),
                         );
                     }
+                    Some(Message::VersionMismatch { version }) => {
+                        // Upstairs cannot communicate with the downstairs.
+                        up.ds_transition(
+                            up_coms.client_id, DsState::BadVersion
+                        ).await;
+                        bail!(
+                            "downstairs version is {}, ours is {}!",
+                            version,
+                            CRUCIBLE_MESSAGE_VERSION,
+                        );
+                    }
                     Some(Message::YesItsMe { version, repair_addr }) => {
                         if negotiated != 0 {
                             bail!("Got version already!");
                         }
 
                         /*
-                         * XXX Valid version to compare with should come
-                         * from main task. In the future we will also have
-                         * to handle a version mismatch.
+                         * For now, we only match on a single version and
+                         * ignore any additional supported versions.
+                         * In the future, we may support a version that
+                         * is different than ours.
                          */
-                        if version != 1 {
-                            up.ds_transition(
-                                up_coms.client_id,
-                                DsState::BadVersion
-                            ).await;
-                            bail!("expected version 1, got {}", version);
+                        match version {
+                            CRUCIBLE_MESSAGE_VERSION => {}
+                            x => {
+                                up.ds_transition(
+                                    up_coms.client_id,
+                                    DsState::BadVersion
+                                ).await;
+                                bail!("expected version {}, got {}",
+                                    CRUCIBLE_MESSAGE_VERSION,
+                                    x
+                                );
```

Comment:
could this be an if/else? the empty match block for `CRUCIBLE_MESSAGE_VERSION` throws me a bit